packages/vite/src/node/config.ts

```diff
@@ -539,7 +541,7 @@ export async function resolveConfig(
   )
 
   // resolve cache directory
-  const pkgPath = lookupFile(resolvedRoot, [`package.json`], { pathOnly: true })
+  const pkgPath = findNearestPackageData(resolvedRoot, packageCache)?.dir
```

Comment:
We are going to be reading the app package json anyways, so there is no perf regression in not using `pathOnly` here.

Reply:
Should we name this `pkgDir`, and use it directly on line 549?

```diff
 const cacheDir = normalizePath(
     config.cacheDir
       ? path.resolve(resolvedRoot, config.cacheDir)
      : pkgPath
-     ? path.join(path.dirname(pkgPath), `node_modules/.vite`)
+     ? path.join(pkgDir, `node_modules/.vite`)
      : path.join(resolvedRoot, `.vite`),
  )
```

Reply:
Good catch! Done